mods/persistence/controllers/subsystems/persistence.dm

```diff
@@ -209,20 +246,13 @@
 			serializer.Commit() // cleanup leftovers.
 
 		// Insert our z-level remaps.
-		var/list/z_inserts = list()
```

Comment:
Why was this moved to the sql serializer? the z_inserts aren't really specific to that serializer and it doesn't reference anything related to the serializer itself. I think the serializer.Commit() and CommitRefUpdates() are actually extraneous and can be removed for the former proc, and moved above the z-level remaps section for the latter proc

Reply:
Because, you put sql code in the subsystem. So if you use another sort of serializer its gonna run sql command when it probably shouldn't. The main goal is to preserver encapsulation. And it also allows to keep defines contained in the serializer code and not exposed everywhere. If its not related to the serializer it shouldn't try to directly interact with the SQL tables that the serializer uses. 

Also, I'm not changing the ordering of things in this pr, or removing existing code. I'm looking to fix the saving issues first and foremost. If there's superfluous stuff, we can handle that later in another PR. I just wanna keep the risks of introducing more problem low. Especially since I'd need to retest everything if I made those changes

Reply:
Not super certain on this still but the functionality is identical so it's okay for now. Note to self to go back and make the z transposing a bit more robust.